language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -811,6 +811,18 @@ private Optional<TypeSymbol> getPositionalArgumentTypeForNewExpr(NodeList<Functi
         Optional<MethodSymbol> methodSymbol = context.currentSemanticModel()
                 .flatMap(semanticModel -> semanticModel.typeOf(newExpressionNode))
                 .flatMap(typeSymbol -> Optional.of(CommonUtil.getRawType(typeSymbol)))
+                .map(typeSymbol -> {
+                    if (typeSymbol.typeKind() == TypeDescKind.UNION) {
+                        Optional<TypeSymbol> classType =
+                                ((UnionTypeSymbol) typeSymbol).memberTypeDescriptors().stream()
+                                        .map(CommonUtil::getRawType)
+                                        .filter(member -> member instanceof ClassSymbol).findFirst();
```

Comment:
Should we handle this here? What if we return the union from here and then ask the corresponding context (ExplicitNewExpressionNodeContext) to do the narrowing down?

Reply:
As per the offline discussion we had here the context type resolver should return the expected type of the positional argument node. Therefore we should not return the union type. We need to narrow it down here and the resolve a posible type symbol for the positional argument.